quarkchain/cluster/tests/test_grpc_client.py

```diff
@@ -5,14 +5,21 @@
 from quarkchain.generated import grpc_pb2
 from quarkchain.generated import grpc_pb2_grpc
 from quarkchain.cluster.grpc_client import GrpcClient
-from quarkchain.core import RootBlockHeader, RootBlock
+from quarkchain.core import RootBlockHeader, RootBlock, MinorBlockHeader
 
 
 class NormalServer(grpc_pb2_grpc.ClusterSlaveServicer):
     def AddRootBlock(self, request, context):
-        return grpc_pb2.AddRootBlockResponse(
-            status=grpc_pb2.ClusterSlaveStatus(code=0, message="Added")
+        grpc_request = grpc_pb2.AddRootBlockRequest(
+            id=request.id,
+            prev_id=request.prev_id,
+            height=request.height,
+            minor_block_headers=request.minor_block_headers,
         )
+        if request.minor_block_headers == grpc_request.minor_block_headers:
```

Comment:
use `self.assertXXX`?

Reply:
It will affect the return value and cause error for normal server... Can I just use the if statements?

Reply:
wait...this check seems will always succeed, right?

Reply:
I thought you want to assert `len(request.minor_block_headers) == 2` since you have it in line 70

Reply:
If an invalid channel was used, the server will not receive a correct root block and this check will not pass.
However, in the error server class, it is just a fake server that returns the invalid response code. It will not reflect on the the block it receives.

Reply:
One more check was added in grpc_client and test for minor block headers. The response will be true if and only if both response code and message, which represents the minor block headers, are correct.